From af213f00ca95b0d04764884dc9d87539e965dd35 Mon Sep 17 00:00:00 2001
From: Sedat Dilek <sedat.dilek@gmail.com>
Date: Thu, 16 Jun 2011 16:11:31 +0200
Subject: [PATCH] Enable IPv6 for Linux kernel smaller 2.6.14

This is a refreshed version of 011-ipv6_oldkernels.patch by Whoopie.

The patch was renamed to clearly indicate its function (as suggested by er13).

An identical include was removed and the include moved to top of file.
Some comments were added and the code was beautified a bit.

Tested IPv6 with my Speedport W 701V box based on v2.6.13.1 Linux-kernel.

Signed-off-by: Sedat Dilek <sedat.dilek@gmail.com>
---
v2: Renamed to 011-ipv6_defines_according_to_kernel_version.patch

 libc/sysdeps/linux/common/bits/in.h |   15 ++++++++++++++-
 1 files changed, 14 insertions(+), 1 deletions(-)

diff --git libc/sysdeps/linux/common/bits/in.h libc/sysdeps/linux/common/bits/in.h
index 1f2b817..51c499f 100644
--- libc/sysdeps/linux/common/bits/in.h
+++ libc/sysdeps/linux/common/bits/in.h
@@ -18,6 +18,8 @@
 
 /* Linux version.  */
 
+#include <linux/version.h>	/* Enable IPv6 for Linux-kernel <=2.6.14.  */
+
 #ifndef _NETINET_IN_H
 # error "Never use <bits/in.h> directly; include <netinet/in.h> instead."
 #endif
@@ -106,13 +108,22 @@ struct in_pktinfo
    The first word in the comment at the right is the data type used;
    "bool" means a boolean value stored in an `int'.  */
 #define IPV6_ADDRFORM		1
+#define IPV6_CHECKSUM		7
+# if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,14)
 #define IPV6_2292PKTINFO	2
 #define IPV6_2292HOPOPTS	3
 #define IPV6_2292DSTOPTS	4
 #define IPV6_2292RTHDR		5
 #define IPV6_2292PKTOPTIONS	6
-#define IPV6_CHECKSUM		7
 #define IPV6_2292HOPLIMIT	8
+# else
+#define IPV6_PKTINFO		2
+#define IPV6_HOPOPTS		3
+#define IPV6_DSTOPTS		4
+#define IPV6_RTHDR		5
+#define IPV6_PKTOPTIONS		6
+#define IPV6_HOPLIMIT		8
+# endif
 
 #define SCM_SRCRT		IPV6_RXSRCRT
 
@@ -134,6 +145,7 @@ struct in_pktinfo
 #define IPV6_IPSEC_POLICY	34
 #define IPV6_XFRM_POLICY	35
 
+# if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,14)
 #define IPV6_RECVPKTINFO	49
 #define IPV6_PKTINFO		50
 #define IPV6_RECVHOPLIMIT	51
@@ -148,6 +160,7 @@ struct in_pktinfo
 
 #define IPV6_RECVTCLASS		66
 #define IPV6_TCLASS		67
+# endif
 
 /* Obsolete synonyms for the above.  */
 #define IPV6_ADD_MEMBERSHIP	IPV6_JOIN_GROUP
-- 
1.7.5.4

